ios/AirPushNotification/AirPushNotification.m

```diff
@@ -139,10 +138,16 @@ void didReceiveRemoteNotification(id self, SEL _cmd, UIApplication* application,
 
 // register the device for push notification.
 DEFINE_ANE_FUNCTION(registerPush)
-{    
-    
-    [[UIApplication sharedApplication] registerForRemoteNotificationTypes:
-     (UIRemoteNotificationTypeBadge | UIRemoteNotificationTypeSound | UIRemoteNotificationTypeAlert)];
+{
+    if ([[UIApplication sharedApplication] respondsToSelector:@selector(registerUserNotificationSettings:)]) {
```

Comment:
Might be worth a comment here that this block is for iOS8, and the else is for legacy systems.